File: src/Application Specific/Classes/Entity Collections/Jobs/JobAudit/JobAuditSQL.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace JobAudits

        Public Class JobAuditSQL


                Private _database As Sys.Database

                Public Sub New(ByVal database As Sys.Database)
                    _database = database
                End Sub

#Region "Functions"

            Public Function [Insert](ByVal oJobAudit As JobAudit) As JobAudit
                _database.ClearParameter()
                _database.AddParameter("@JobID", oJobAudit.JobID, True)
                _database.AddParameter("@ActionChange", oJobAudit.ActionChange, True)
                _database.AddParameter("@ActionDateTime", Now, True)
                _database.AddParameter("@UserID", loggedInUser.UserID, True)

                oJobAudit.SetJobAuditID = Entity.Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("JobAudit_Insert"))
                Return oJobAudit
            End Function

            Public Function [Insert](ByVal oJobAudit As JobAudit, ByVal trans As SqlClient.SqlTransaction) As JobAudit

                Dim Command As SqlClient.SqlCommand = New SqlClient.SqlCommand

                Command.CommandText = "JobAudit_Insert"
                Command.CommandType = CommandType.StoredProcedure
                Command.Transaction = trans
                Command.Connection = trans.Connection

                Command.Parameters.Add(New System.Data.SqlClient.SqlParameter("@JobID", oJobAudit.JobID))
                Command.Parameters.Add(New System.Data.SqlClient.SqlParameter("@ActionChange", oJobAudit.ActionChange))
                Command.Parameters.Add(New System.Data.SqlClient.SqlParameter("@ActionDateTime", Now))
                Command.Parameters.Add(New System.Data.SqlClient.SqlParameter("@UserID", loggedInUser.UserID))

                oJobAudit.SetJobAuditID = Entity.Sys.Helper.MakeIntegerValid(Command.ExecuteScalar)
                Return oJobAudit

            End Function

            Public Function Get_For_Job(ByVal JobID As Integer) As DataView
                _database.ClearParameter()
                _database.AddParameter("@JobID", JobID, True)

                Dim dt As DataTable = _database.ExecuteSP_DataTable("JobAudit_Get")
                dt.TableName = Entity.Sys.Enums.TableNames.tblJobAudit.ToString
                Return New DataView(dt)
            End Function

#End Region

            End Class

        End Namespace

End Namespace

File: apollo/apollo.fsm/src/Application Specific/Classes/Entity Collections/Jobs/JobAudit/JobAuditSQL.cs
using System.Data;
using System.Data.SqlClient;
using Microsoft.VisualBasic;

namespace FSM.Entity
{
    namespace JobAudits
    {
        public class JobAuditSQL
        {
            private Sys.Database _database;

            public JobAuditSQL(Sys.Database database)
            {
                _database = database;
            }

            
            public JobAudit Insert(JobAudit oJobAudit)
            {
                _database.ClearParameter();
                _database.AddParameter("@JobID", oJobAudit.JobID, true);
                _database.AddParameter("@ActionChange", oJobAudit.ActionChange, true);
                _database.AddParameter("@ActionDateTime", DateAndTime.Now, true);
                _database.AddParameter("@UserID", App.loggedInUser.UserID, true);
                oJobAudit.SetJobAuditID = Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("JobAudit_Insert"));
                return oJobAudit;
            }

            public JobAudit Insert(JobAudit oJobAudit, SqlTransaction trans)
            {
                var Command = new SqlCommand();
                Command.CommandText = "JobAudit_Insert";
                Command.CommandType = CommandType.StoredProcedure;
                Command.Transaction = trans;
                Command.Connection = trans.Connection;
                Command.Parameters.Add(new SqlParameter("@JobID", oJobAudit.JobID));
                Command.Parameters.Add(new SqlParameter("@ActionChange", oJobAudit.ActionChange));
                Command.Parameters.Add(new SqlParameter("@ActionDateTime", DateAndTime.Now));
                Command.Parameters.Add(new SqlParameter("@UserID", App.loggedInUser.UserID));
                oJobAudit.SetJobAuditID = Sys.Helper.MakeIntegerValid(Command.ExecuteScalar());
                return oJobAudit;
            }

            public DataView Get_For_Job(int JobID)
            {
                _database.ClearParameter();
                _database.AddParameter("@JobID", JobID, true);
                var dt = _database.ExecuteSP_DataTable("JobAudit_Get");
                dt.TableName = Sys.Enums.TableNames.tblJobAudit.ToString();
                return new DataView(dt);
            }

            
        }
    }
}